buildscripts/autogen_sources.py

```diff
@@ -0,0 +1,155 @@
+#
+#     Redistributions of source code must retain the above copyright notice,
+#     this list of conditions and the following disclaimer.
+#
+#     Redistributions in binary form must reproduce the above copyright notice,
+#     this list of conditions and the following disclaimer in the documentation
+#     and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+# THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+# OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+# WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+# OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+# EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+# *****************************************************************************
+'''
+
+docstring_header = '''\
+"""
+
+| This file contains SDC Pandas Series methods auto-generated with autogen_sources.py
+
+"""
+'''
+
+arithmetic_binops_symbols = {
+    'add': '+',
+    'sub': '-',
+    'mul': '*',
+    'truediv': '/',
+    'floordiv': '//',
+    'mod': '%',
+    'pow': '**',
+}
+
+comparison_binops_symbols = {
+    'lt': '<',
+    'gt': '>',
+    'le': '<=',
+    'ge': '>=',
+    'ne': '!=',
+    'eq': '==',
+}
+
+target_rel_filename = 'sdc/datatypes/hpat_pandas_series_autogenerated.py'
+
+
+if __name__ == '__main__':
+    sdc_src = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+    abs_file_name = os.path.join(sdc_src, target_rel_filename)
+
+    # read templates_module as text and extract import section to be copied into auto-generated target file
+    module_text = inspect.getsource(templates_module)
+    module_text_lines = module_text.splitlines(keepends=True)
+    imports_line_numbers = [k for k, line in enumerate(module_text_lines) if 'import ' in line]
+    imports_start_line, import_end_line = min(imports_line_numbers), max(imports_line_numbers)
+    import_section_text = ''.join(module_text_lines[imports_start_line: import_end_line + 1])
+
+    # read function templates for arithmetic and comparison operators from tempaltes module
+    func_template_arithmetic_binop = inspect.getsource(templates_module.sdc_pandas_series_operator_binop)
+    func_template_comparison_binop = inspect.getsource(templates_module.sdc_pandas_series_operator_comp_binop)
+
+    exit_status = -1
+    try:
+        # open the target file for writing and do the main work
+        with open(abs_file_name, 'w', newline='') as file:
+            file.write(encoding_info)
+            file.write('\n')
```

Comment:
You could merge such lines of code:
`file.write(f'{encoding_info}\n')`

Reply:
Yes, thank you!